### ISSUE TITLE
Experimental feature failure -  unknown variable accessed

### ISSUE THREAD

**Issue by myoung34** [hashicorp/terraform#9925]
### Terraform Version
0.7.9

### Affected Resource(s)
Please list the resources as a list, for example:
- module
- aws_ecs_cluster
- aws_ecs_service

### Terraform Configuration Files
```
  module "ecs_cluster" {
    source = "../../../../modules/ecs_cluster"

    subnet_id = "subnet-..."
    ecs_cluster_name = "health-staging"
    security_group_id = "sg-..."
    key_name = "health"
    ecs_sigterm_timeout = "45m"
    environment = "${var.environment}"
  }

  module "ecs_service_reports_blackbird" {
    source = "../../../../modules/ecs_service"

    ecs_cluster_id = "${module.ecs_cluster.cluster_id}"
    worker_count = 8
    worker_name = "reports-blackbird"
    worker_command = "python manage.py rqworker reports blackbird"
    vault_host = "${var.vault_host}"
    vault_token = "${var.vault_token}"
    environment = "${var.environment}"
    health_env = "${var.health_env}"
    cpu = "${var.cpu}",
    memory = "${var.memory}",
    memory_reserved = "${var.memory_reserved}"
  }
```

cluster module `main.tf` (reduced):

```
  resource "aws_ecs_cluster" "main" {
      name = "${var.ecs_cluster_name}"
  }
```

cluster module `outputs.tf`:

```
output "cluster_id" {
  value = "${aws_ecs_cluster.main.id}"
}
```

service module `main.tf` (reduced)

```
  resource "aws_ecs_service" "health-worker" {
    name                               = "health-${var.worker_name}-${var.environment}"
    cluster                            = "${var.ecs_cluster_id}"
    task_definition                    = "${aws_ecs_task_definition.health-worker.arn}"
    desired_count                      = "${var.worker_count}"
    deployment_maximum_percent         = "${var.deployment_maximum_percent}"
    deployment_minimum_healthy_percent = "${var.deployment_minimum_healthy_percent}"
  }
```

### Debug Output
https://gist.github.com/myoung34/b1cd5a7dbde5e46f5b98eb008e9338bd

### Expected Behavior
Create the cluster then successfully create the service

### Actual Behavior
Threw warnings: 

https://gist.github.com/myoung34/b1cd5a7dbde5e46f5b98eb008e9338bd

### Steps to Reproduce
Please list the steps required to reproduce the issue, for example:
1. `terraform apply`

### Important Factoids
Tight VPC, private instances

**Comment by paddyforan** [comment]
Hi @myoung34! Just to clarify: did the resources get created correctly, and you're just reporting the warnings? Or did the resources not get created as expected?

**Comment by myoung34** [comment]
They did create successfully and everything is fine, it just doesn't seem like they did according to the error log

**Comment by myoung34** [comment]
Also to add more information this did not happen in `0.7.5` which was the version I upgraded from to `0.7.9`. After upgrading this began to show.

**Comment by paddyforan** [comment]
Ah! Then you've run into a shadow graph bug.

So, the good news is: there's nothing to worry about at the moment! Everything is fine, terraform is behaving as it should.

The better news is: you've helped us preemptively find a bug in an experimental feature of terraform that we're testing out. The important part of that message is here: https://gist.github.com/myoung34/b1cd5a7dbde5e46f5b98eb008e9338bd#file-gistfile1-txt-L1-L15

We'll be looking at fixing the bug you stumbled on, but for now, the workaround is to just ignore the message. :)

Sorry for all the confusion. We'll talk about how to make this clearer in the future. In the meantime, if you experience any more problems (or see any more of these messages!) please feel free to open issues.

**Comment by mitchellh** [comment]
This is fixed in master :) will be in 0.7.10
